libmamba/src/core/shell_init.cpp

```diff
@@ -49,7 +51,8 @@ namespace mamba
         {
             return "zsh";
         }
-        if (contains(parent_process_name, "xonsh"))
+        // xonsh in unix, Python in macOS
+        if (contains(parent_process_name, "xonsh") || contains(parent_process_name, "Python"))
```

Comment:
Hmm, is this safe? What if we run `subprocess.run(["micromamba", "shell", "init"])`, then the parent process is Python.

I think it would be ok to require explicit `-s xonsh` instead.

Reply:
My python parent process is called `python3.10`. Since micromamba looks for `Python` with a capital P, everything should be fine. 🤡🤪 

I think a good solution would be to check for python (case insensitive) and print a special info message telling the user about `-s xonsh`.